cpp/src/arrow/flight/client.cc

```diff
@@ -51,6 +51,7 @@
 
 #include "arrow/flight/client_auth.h"
 #include "arrow/flight/client_middleware.h"
+#include "arrow/flight/client_header_auth_middleware.h"
```

Comment:
Is this ordering alphabetical? If so this should be up one...

Reply:
Yes, was waiting for feedback on design before proper formatting. I will correct this.